src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentTransactionInfoPlugin.java

```diff
@@ -156,7 +139,33 @@ public PaymentPluginStatus getStatus() {
         }
     }
 
-    private static PaymentPluginStatus getPaymentPluginStatus(final PaymentServiceProviderResult pspResult) {
+    /**
+     * Transforms adyenCallErrorStatus (where there any technical errors?) and pspResult (was the call successful from a business perspective) into the PaymentPluginStatus.
+     * Therefor
+     */
+    private static PaymentPluginStatus getPaymentPluginStatus(final Optional<AdyenCallErrorStatus> adyenCallErrorStatus, final Optional<PaymentServiceProviderResult> pspResult) {
+        checkArgument(adyenCallErrorStatus.isPresent() ^ pspResult.isPresent());
+        return (pspResult.isPresent()) ? pspResultToPaymentPluginStatus(pspResult.get()) : adyenCallErrorStatusToPaymentPluginStatus(adyenCallErrorStatus.get());
+    }
+
+    private static PaymentPluginStatus adyenCallErrorStatusToPaymentPluginStatus(final AdyenCallErrorStatus adyenCallErrorStatus) {
+        switch (adyenCallErrorStatus) {
+            case REQUEST_NOT_SEND:
+                return PaymentPluginStatus.CANCELED;
+            case RESPONSE_ABOUT_INVALID_REQUEST:
+                return PaymentPluginStatus.ERROR;
+            case RESPONSE_NOT_RECEIVED:
```

Comment:
yeah i did that in the first place and decided later to be more specific (even if the code is now longer). The reason is mainly that the whole stuff is a best guess approach, so it makes sense to explicit note all cases down which are identified. If someone later has to add exceptions he knows if it was already identified (and maybe just mapped wrong) or if it's a new case.

Once we have some real production data we can adjust/simplify the method.